packages/jest-snapshot/src/inline_snapshots.ts

```diff
@@ -60,43 +54,82 @@ const saveSnapshotsForFile = (
   snapshots: Array<InlineSnapshot>,
   sourceFilePath: Config.Path,
   prettier: any,
-  babelTraverse: Function,
+  babelTraverse: BabelTraverse,
 ) => {
   const sourceFile = fs.readFileSync(sourceFilePath, 'utf8');
 
-  // Resolve project configuration.
-  // For older versions of Prettier, do not load configuration.
-  const config = prettier.resolveConfig
-    ? prettier.resolveConfig.sync(sourceFilePath, {
-        editorconfig: true,
-      })
-    : null;
-
-  // Detect the parser for the test file.
-  // For older versions of Prettier, fallback to a simple parser detection.
-  const inferredParser = prettier.getFileInfo
-    ? prettier.getFileInfo.sync(sourceFilePath).inferredParser
-    : (config && config.parser) || simpleDetectParser(sourceFilePath);
-
-  // Insert snapshots using the custom parser API. After insertion, the code is
-  // formatted, except snapshot indentation. Snapshots cannot be formatted until
-  // after the initial format because we don't know where the call expression
-  // will be placed (specifically its indentation).
-  const newSourceFile = prettier.format(sourceFile, {
-    ...config,
-    filepath: sourceFilePath,
-    parser: createInsertionParser(snapshots, inferredParser, babelTraverse),
-  });
+  const {options} = loadPartialConfig({filename: sourceFilePath})!;
+  if (!options.plugins) {
+    options.plugins = [];
+  }
 
-  // Format the snapshots using the custom parser API.
-  const formattedNewSourceFile = prettier.format(newSourceFile, {
-    ...config,
-    filepath: sourceFilePath,
-    parser: createFormattingParser(inferredParser, babelTraverse),
-  });
+  // TypeScript projects may not have a babel config; make sure they can be parsed anyway.
+  if (/\.tsx?$/.test(sourceFilePath)) {
+    options.plugins.push('typescript');
```

Comment:
do we depend on these plugins? If not, we should. Also, maybe do `require.resolve` on them so we pass a full path to `babel`?

What happens if the user has `typescript` in there already? Do we risk double?

Reply:
> do we depend on these plugins? If not, we should. Also, maybe do require.resolve on them so we pass a full path to babel?

I got both `typescript` and `jsx` from the official babel parser docs: https://babeljs.io/docs/en/babel-parser#language-extensions. They’re also in [the typescript definition file](https://github.com/babel/babel/blob/1d3f9815df747902ae535706f032081bc4825b87/packages/babel-parser/typings/babel-parser.d.ts#L93-L96).
A `require.resolve` wouldn’t work here because, as I understand it, `typescript` is a special babel plugin name referring to typescript. `require.resolve(‘typescript’)` would return the `typescript` node module itself.
See my other comment for some simple equivalent code runnable with only a jest install (i.e. no explicit install of babel, typescript, or any plugins): https://github.com/facebook/jest/pull/7792/files#r298783449

> Do we risk double?

Good point - from having tested manually, nothing blows up when there’s a duplicate plugin, but I’ve added a check to make sure the plugins aren’t already present before adding them anyway.

Reply:
I meant `require.resolve('@babel/plugin-transform-typescript')`

Reply:
I do think this needs to be addressed, or it'll break horribly in e.g. yarn pnp